src/services/formatting/formatting.ts

```diff
@@ -387,7 +389,10 @@ namespace ts.formatting {
 
             let indentation = inheritedIndentation;
             if (indentation === Constants.Unknown) {
-                if (isSomeBlock(node.kind)) {
+                if (isIndentPreventedChildNode(parent.kind, node.kind)) {

```

Comment:
possibly move getIndentationForToken to a new helper, call it in getIndentationForToken, and from here and add support for jsx elements in it as well.. possibly give the new helper a better name as getIndentationForNode.